CMakeLists.txt

```diff
@@ -126,6 +130,8 @@ if (BUILD_STATIC_RELEASE)
   # Do not bring in dynamic libstdcc and libgcc
   set(CMAKE_EXE_LINKER_FLAGS "-static -static-libgcc -static-libstdc++ -Wl,-z,muldefs")
   add_definitions(-DP4C_STATIC_BUILD)
+else ()
+  set(BUILD_SHARED_LIBS ON)
```

Comment:
Internal libraries are already linked statically since we only build them this way, e.g. 
https://github.com/p4lang/p4c/blob/main/ir/CMakeLists.txt#L66

The intention was to allow Protobuf to be linked dynamically to reduce binary size as per the discussion in #4055. But it is probably more prudent to always package it in to avoid lookup issues even though that may mean we get a huge binary.

Reply:
I see. How much does the protobuf increase binary size? But I would still suggest using static protobuf.